src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -698,27 +678,27 @@ private static void UpdateSymbolCacheNoLock(List<int> newSymbols, ImmutableArray
             }
 
             private static void SetupReferencesForSourceAssembly(
+                ImmutableArray<AssemblyData> allAssemblyData,
```

Comment:
It looks like we are using it only for length. Consider passing allAssemblyData.Length - 1 instead.